__tests__/PropTypesDevelopmentReact15.js

```diff
@@ -875,6 +875,89 @@ describe('PropTypesDevelopmentReact15', () => {
     });
   });
 
+  describe('ElementWithType Types', () => {
+    it('should fail for mismatched component.type', () => {
+      typeCheckFail(
+        PropTypes.elementWithType('div'),
+        <span/>,
+        "Warning: Failed prop type: Invalid prop `testProp` of type [span] supplied to `testComponent`, expected one of type `div`."
+      );
+    });
+
+    it('should pass for matched component.type', () => {
+      typeCheckPass(PropTypes.elementWithType('div'), <div />);
+    });
+
+    it('should pass if no component given', () => {
```

Comment:
This needs a test case for the required version that fails

Reply:
updated. I'm adding a task to cover the "syching specs" part.